### PR TITLE
Mail templates - add descriptions to password recovery templates

### PR DESCRIPTION
**Pull request by dsturm**
### 1. Why is this change necessary?

Currently, due to the naming / translation of the types for the two password recovery templates (Password recovery), (in German) it is unclear which template is used for which addressee.

grafik

### 2. What does this change do, exactly?

Therefore, this PR adds a description to distinguish them. It also corrects a small typo in the German translation of the mail subject.

### 3. Describe each step to reproduce the issue or behaviour.

1. Login into backend
2. Go to `Settings` > `Shop` > `Email Templates`
3. (optional) Switch to German

### 5. Checklist

- ~[ ] I have written tests and verified that they fail without my change~
- [x] I have squashed any insignificant commits
- ~[ ] I have created a [changelog file](https://github.com/shopware/platform/blob/trunk/adr/2020-08-03-Implement-New-Changelog.md) with all necessary information about my changes~
- ~[ ] I have written or adjusted the documentation according to my changes~
- ~[ ] This change has comments for package types, values, functions, and non-obvious lines of code~
- [x] I have read the contribution requirements and fulfil them.

### CODE REVIEW

File: src/Core/Migration/V6_3/Migration1562240231UserPasswordRecovery.php [hunk 1]
```diff
@@ -111,7 +111,7 @@ private function createMailTemplate(Connection $connection, string $mailTemplate
                 'language_id' => $defaultLangId,
                 'sender_name' => 'Shopware Administration',
                 'subject' => 'Password recovery',
-                'description' => '',
+                'description' => 'Mail to reset password for backend users',
```
Comment (mitelg) [inline src/Core/Migration/V6_3/Migration1562240231UserPasswordRecovery.php]:
```suggestion
                'description' => 'Mail to reset password for administration users',
```

File: src/Core/Migration/V6_3/Migration1562240231UserPasswordRecovery.php [hunk 2]
```diff
@@ -124,7 +124,7 @@ private function createMailTemplate(Connection $connection, string $mailTemplate
                 'language_id' => Uuid::fromHexToBytes(Defaults::LANGUAGE_SYSTEM),
                 'sender_name' => 'Shopware Administration',
                 'subject' => 'Password recovery',
-                'description' => '',
+                'description' => 'Mail to reset password for backend users',
```
Comment (mitelg) [inline src/Core/Migration/V6_3/Migration1562240231UserPasswordRecovery.php]:
```suggestion
                'description' => 'Mail to reset password for administration users',
```

File: src/Core/Migration/V6_3/Migration1562240231UserPasswordRecovery.php [hunk 3]
```diff
@@ -136,8 +136,8 @@ private function createMailTemplate(Connection $connection, string $mailTemplate
                 'mail_template_id' => Uuid::fromHexToBytes($mailTemplateId),
                 'language_id' => $deLangId,
                 'sender_name' => 'Shopware Administration',
-                'subject' => 'Password Wiederherstellung',
-                'description' => '',
+                'subject' => 'Passwort Wiederherstellung',
+                'description' => 'Mail zum Zurücksetzen des Passworts für Backend Nutzer',
```
Comment (mitelg) [inline src/Core/Migration/V6_3/Migration1562240231UserPasswordRecovery.php]:
```suggestion
                'description' => 'Mail zum Zurücksetzen des Passworts für Adminstations-Nutzer',
```

**Comment by MalteJanz** [comment]
Hi, thank you very much for your contribution. I would like to accept it but unfortunately there is a fatal flaw in the current implementation:

It changes old database migrations, so only new users (which haven't run these migrations yet - haven't updated shopware) would benefit from this change. Also the modified migrations are quite old.

The right way to implement it would be a new migration which updates the email templates. It would be nice if you could refactor your implementation and if you need any mentoring / help during the process let me know.

To get you started have a look at how more recent migrations update the email templates, for example this one:
https://github.com/shopware/platform/blob/trunk/src/Core/Migration/V6_4/Migration1636014089UpdateOrderConfirmationMailTemplates.php

There is also quite a bit of documentation regarding writing database migrations:
- (this one is for plugins but most of it also applies here) https://developer.shopware.com/docs/guides/plugins/plugins/plugin-fundamentals/database-migrations
- the docs also link to this great video tutorial about the topic: https://youtu.be/__pWwaK6lxw

**Comment by dsturm** [comment]
Hey @MalteJanz, sure - my mistake. I should have asked this as a question right away. 🤦🏼‍♂️ I'll try to update this PR as soon as possible and add a migration for the changes.

**Comment by MalteJanz** [comment]
Hi @dsturm , as it seems like the progress on this PR has stopped and the implementation is far from done I will close this PR for now.

Feel free to create a new PR in the future to solve the problem with a new database migration. Have a nice day 🙂.

**Pull request closed without merging**